ocaml/xapi/certificates.ml

```diff
@@ -74,100 +75,110 @@ let not_safe_chars name =
   in
   f 0
 
-let not_safe is_cert name =
+let is_unsafe kind name =
   name.[0] = '.' ||
-  not (String.endswith ".pem" name) ||
+  not (Astring.String.is_suffix ~affix:".pem" name) ||
   not_safe_chars name
 
-let raise_server_error n err =
-  raise (Server_error(err, [n]))
+let raise_server_error parameters err =
+  raise (Server_error(err, parameters))
 
-let raise_name_invalid is_cert n =
-  raise_server_error n
-    (if is_cert then certificate_name_invalid else crl_name_invalid)
+let raise_name_invalid kind n =
+  let err = match kind with
+  | Certificate -> certificate_name_invalid
+  | CRL -> crl_name_invalid
+  in
+  raise_server_error [n] err
 
-let raise_already_exists is_cert n =
-  raise_server_error n
-    (if is_cert then certificate_already_exists else crl_already_exists)
+let raise_already_exists kind n =
+  let err = match kind with
+  | Certificate -> certificate_already_exists
+  | CRL -> crl_already_exists
+  in
+  raise_server_error [n] err
 
-let raise_does_not_exist is_cert n =
-  raise_server_error n
-    (if is_cert then certificate_does_not_exist else crl_does_not_exist)
+let raise_does_not_exist kind n =
+  let err = match kind with
+  | Certificate -> certificate_does_not_exist
+  | CRL -> crl_does_not_exist
+  in
+  raise_server_error [n] err
 
-let raise_corrupt is_cert n =
-  raise_server_error n
-    (if is_cert then certificate_corrupt else crl_corrupt)
+let raise_corrupt kind n =
+  let err = match kind with
+  | Certificate -> certificate_corrupt
+  | CRL -> crl_corrupt
+  in
+  raise_server_error [n] err
 
 let raise_library_corrupt () =
   raise (Server_error (certificate_library_corrupt, []))
 
-let local_list is_cert =
-  mkdir_cert_path is_cert;
+let local_list kind =
+  mkdir_cert_path kind;
   List.filter
     (fun n ->
-       let stat = Unix.lstat (library_filename is_cert n) in
+       let stat = Unix.lstat (library_filename kind n) in
        stat.Unix.st_kind = Unix.S_REG)
-    (Array.to_list (Sys.readdir (library_path is_cert)))
+    (Array.to_list (Sys.readdir (library_path kind)))
 
 let local_sync () =
   try
     rehash()
-  with
-  | e ->
+  with e ->
     warn "Exception rehashing certificates: %s"
       (ExnHelper.string_of_exn e);
     raise_library_corrupt()
 
-let cert_perms is_cert =
-  let stat = Unix.stat (library_path is_cert) in
-  debug "%d %d" (stat.Unix.st_perm land 0o666) stat.Unix.st_perm;
-  stat.Unix.st_perm land 0o666
-
-let host_install is_cert ~name ~cert =
-  if not_safe is_cert name then
-    raise_name_invalid is_cert name;
-  let filename = library_filename is_cert name in
+let cert_perms kind =
+  let stat = Unix.stat (library_path kind) in
+  let mask = 0o666 in
+  let perm = stat.Unix.st_perm land mask in
+  debug "%d %d" perm stat.Unix.st_perm;
```

Comment:
Has this log line enough context to be useful? These are just two numbers.

Reply:
I don't think so, I'm partial to removing the logline.

I changed this function because it's most probably going to be reused for the new certificates, which need different permissions.